consensus/core/src/leader_schedule.rs

```diff
@@ -114,6 +114,14 @@ impl LeaderSchedule {
             .unwrap() as usize
     }
 
+    /// Checks whether the dag state unscored sub dags list is empty. If yes then that means that
+    /// either (1) the system has just started and there is no unscored sub dag available (2) the
+    /// schedule has updated - new scores have been calculated. Both cases we consider as valid cases
+    /// where the schedule has been updated.
```

Comment:
I'm not sure (1) is a valid reason. DagState has to fully recover unscored commits before leader schedule can start to make decisions. Otherwise there can be a fork. Or are you seeing something else when starting consensus?

Reply:
Yes, but if it's a fresh system  then there are not sub dags to recover.